compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -527,7 +527,7 @@ class Inliner(call: tpd.Tree, rhs: tpd.Tree)(implicit ctx: Context) {
       cond1.tpe.widenTermRefExpr match {
         case ConstantType(Constant(condVal: Boolean)) =>
           val selected = typed(if (condVal) tree.thenp else tree.elsep, pt)
-          if (isIdempotentExpr(cond1)) selected
+          if (isPureExpr(cond1)) selected
```

Comment:
That lazy val creates the idempotent condition with constant type. Just found another bug with lazy vals with constant type #2266.